### PR TITLE
You will no longer burn your hand trying to pick something up below a lightbulb because you clicked a near invisible pixel.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request
You will no longer burn your hand trying to pick something up below a lightbulb because you clicked a near invisible pixel.

## Why It's Good For The Game

Thousands of times over the years, I have taken 5 burn damage because I tried to click on an item and accidentally clicked these dumb goddamn 90% transparent pixels. Fuck this shit so much.

## Changelog
:cl:
imageadd: You will no longer burn your hand trying to pick something up below a lightbulb because you clicked a near invisible pixel.
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
draft this PR please i need to finish the rest of these tomorrow

**Comment by timothymtorres** [comment]
This will fuckup Multi-Z pretty bad.  `emissive_appearance` bleeds through multi z.  See - https://github.com/tgstation/tgstation/issues/65800

File: code/modules/power/lighting/light.dm [hunk 1]
```diff
@@ -129,12 +129,12 @@
 
 	var/area/local_area = get_area(src)
 	if(emergency_mode || (local_area?.fire))
-		. += mutable_appearance(overlay_icon, "[base_state]_emergency")
+		. += emissive_appearance(overlay_icon, "[base_state]_emergency")
```
Comment (Fikou) [inline code/modules/power/lighting/light.dm:132]: have you tested this? emissive appearances don't even have color

**Comment by nemvar** [comment]
Logged back in just to talk about this.
This shouldn't work.
From the DM reference  /atom/var/mouse_opacity

> Note that overlays and underlays are not distinct objects, so their mouse_opacity is completely ignored in favor of the object they're attached to. The same applies to [image objects](https://www.byond.com/docs/ref/info.html#/image), which act like special overlays as well. [Visual contents](https://www.byond.com/docs/ref/info.html#/atom/var/vis_contents), on the other hand, are separate objects that can act like overlays in some ways, but because they're separate their mouse_opacity is taken into account.

So if byond is even remotely sane, setting mouse_opacity on the plane master should also not change how overlays of said plane behave.
The only solution I'm seeing visual contents. Useful flags are VIS_INHERIT_DIR and VIS_INHERIT_LAYER.

**Comment by LemonInTheDark** [comment]
> Logged back in just to talk about this. This shouldn't work. From the DM reference /atom/var/mouse_opacity
> 
> > Note that overlays and underlays are not distinct objects, so their mouse_opacity is completely ignored in favor of the object they're attached to. The same applies to [image objects](https://www.byond.com/docs/ref/info.html#/image), which act like special overlays as well. [Visual contents](https://www.byond.com/docs/ref/info.html#/atom/var/vis_contents), on the other hand, are separate objects that can act like overlays in some ways, but because they're separate their mouse_opacity is taken into account.
> 
> So if byond is even remotely sane, setting mouse_opacity on the plane master should also not change how overlays of said plane behave. The only solution I'm seeing is visual contents. Useful flags are VIS_INHERIT_DIR and VIS_INHERIT_LAYER.

:))))))))))
It does. this is a bad fix for other reasons, but it does. planed visual elements get drawn to slates, and those slates decide how the slates will render. so you can modify mouse opacity of overlays

image

Good to see you tho, hope you're doin well

**Pull request closed without merging**